`serialise` benchmarks fail to compile with GHC 9.0 or later

```
$ cabal build bench:micro bench:versus --enable-tests --enable-benchmarks -w ghc-9.0.2
Build profile: -w ghc-9.0.2 -O1
In order, the following will be built (use -v for more details):
 - serialise-0.2.4.0 (bench:micro) (first run)
 - serialise-0.2.4.0 (bench:versus) (first run)
Preprocessing benchmark 'micro' for serialise-0.2.4.0..
Preprocessing benchmark 'versus' for serialise-0.2.4.0..
Building benchmark 'micro' for serialise-0.2.4.0..
Building benchmark 'versus' for serialise-0.2.4.0..
[ 6 of 14] Compiling Micro.PkgAesonTH ( bench/micro/Micro/PkgAesonTH.hs, /home/rscott/Documents/Hacking/Haskell/cborg/dist-newstyle/build/x86_64-linux/ghc-9.0.2/serialise-0.2.4.0/b/micro/build/micro/micro-tmp/Micro/PkgAesonTH.o, /home/rscott/Documents/Hacking/Haskell/cborg/dist-newstyle/build/x86_64-linux/ghc-9.0.2/serialise-0.2.4.0/b/micro/build/micro/micro-tmp/Micro/PkgAesonTH.dyn_o )
[ 6 of 15] Compiling Macro.PkgAesonTH ( bench/versus/Macro/PkgAesonTH.hs, /home/rscott/Documents/Hacking/Haskell/cborg/dist-newstyle/build/x86_64-linux/ghc-9.0.2/serialise-0.2.4.0/b/versus/build/versus/versus-tmp/Macro/PkgAesonTH.o, /home/rscott/Documents/Hacking/Haskell/cborg/dist-newstyle/build/x86_64-linux/ghc-9.0.2/serialise-0.2.4.0/b/versus/build/versus/versus-tmp/Macro/PkgAesonTH.dyn_o )

bench/micro/Micro/PkgAesonTH.hs:12:13: error:
    • No instance for (ToJSON Tree) arising from a use of ‘encode’
      There are instances for similar types:
        instance ToJSON v => ToJSON (containers-0.6.4.1:Data.Tree.Tree v)
          -- Defined in ‘aeson-2.0.3.0:Data.Aeson.Types.ToJSON’
    • In the expression: encode
      In an equation for ‘serialise’: serialise = encode
   |
12 | serialise = Aeson.encode
   |             ^^^^^^^^^^^^

bench/micro/Micro/PkgAesonTH.hs:15:26: error:
    • No instance for (FromJSON Tree) arising from a use of ‘decode'’
      There are instances for similar types:
        instance FromJSON v =>
                 FromJSON (containers-0.6.4.1:Data.Tree.Tree v)
          -- Defined in ‘aeson-2.0.3.0:Data.Aeson.Types.FromJSON’
    • In the second argument of ‘(.)’, namely ‘decode'’
      In the expression: fromJust . decode'
      In an equation for ‘deserialise’: deserialise = fromJust . decode'
   |
15 | deserialise = fromJust . Aeson.decode'
   |                          ^^^^^^^^^^^^^

bench/versus/Macro/PkgAesonTH.hs:12:18: error:
    • No instance for (ToJSON GenericPackageDescription)
        arising from a use of ‘encode’
    • In the expression: encode pkgs
      In an equation for ‘serialise’: serialise pkgs = encode pkgs
   |
12 | serialise pkgs = Aeson.encode pkgs
   |                  ^^^^^^^^^^^^^^^^^

bench/versus/Macro/PkgAesonTH.hs:15:26: error:
    • No instance for (FromJSON GenericPackageDescription)
        arising from a use of ‘decode'’
    • In the second argument of ‘(.)’, namely ‘decode'’
      In the expression: fromJust . decode'
      In an equation for ‘deserialise’: deserialise = fromJust . decode'
   |
15 | deserialise = fromJust . Aeson.decode'
   |                          ^^^^^^^^^^^^^
cabal: Failed to build bench:micro from serialise-0.2.4.0.
Failed to build bench:versus from serialise-0.2.4.0.
```

This can likely be fixed by changing the order of TH splices.

--- a/.github/workflows/haskell-ci.yml
+++ b/.github/workflows/haskell-ci.yml
@@ -116,8 +116,8 @@ jobs:
 
           HCNUMVER=$(${HC} --numeric-version|perl -ne '/^(\d+)\.(\d+)\.(\d+)(\.(\d+))?$/; print(10000 * $1 + 100 * $2 + ($3 == 0 ? $5 != 1 : $3))')
           echo "HCNUMVER=$HCNUMVER" >> "$GITHUB_ENV"
-          if [ $((HCNUMVER < 90000)) -ne 0 ] ; then echo "ARG_TESTS=--enable-tests" >> "$GITHUB_ENV" ; else echo "ARG_TESTS=--disable-tests" >> "$GITHUB_ENV" ; fi
-          if [ $((HCNUMVER < 90000)) -ne 0 ] ; then echo "ARG_BENCH=--enable-benchmarks" >> "$GITHUB_ENV" ; else echo "ARG_BENCH=--disable-benchmarks" >> "$GITHUB_ENV" ; fi
+          echo "ARG_TESTS=--enable-tests" >> "$GITHUB_ENV"
+          echo "ARG_BENCH=--enable-benchmarks" >> "$GITHUB_ENV"
           echo "HEADHACKAGE=false" >> "$GITHUB_ENV"
           echo "ARG_COMPILER=--$HCKIND --with-compiler=$HC" >> "$GITHUB_ENV"
           echo "GHCJSARITH=0" >> "$GITHUB_ENV"
@@ -247,7 +247,7 @@ jobs:
           $CABAL v2-build $ARG_COMPILER $ARG_TESTS $ARG_BENCH all --write-ghc-environment-files=always
       - name: tests
         run: |
-          if [ $((HCNUMVER < 90000)) -ne 0 ] ; then $CABAL v2-test $ARG_COMPILER $ARG_TESTS $ARG_BENCH all --test-show-details=direct ; fi
+          $CABAL v2-test $ARG_COMPILER $ARG_TESTS $ARG_BENCH all --test-show-details=direct
       - name: cabal check
         run: |
           cd ${PKGDIR_cborg} || false

--- a/cabal.haskell-ci
+++ b/cabal.haskell-ci
@@ -1,4 +1,1 @@
 jobs-selection: any
--- These don't yet build with GHC 9.0
-tests: < 9.0
-benchmarks: < 9.0

--- a/serialise/bench/micro/Micro/PkgAesonTH.hs
+++ b/serialise/bench/micro/Micro/PkgAesonTH.hs
@@ -8,11 +8,10 @@ import Data.Aeson.TH as Aeson
 import Data.ByteString.Lazy as BS
 import Data.Maybe
 
+deriveJSON defaultOptions ''Tree
+
 serialise :: Tree -> BS.ByteString
 serialise = Aeson.encode
 
 deserialise :: BS.ByteString -> Tree
 deserialise = fromJust . Aeson.decode'
-
-deriveJSON defaultOptions ''Tree
-

--- a/serialise/bench/versus/Macro/PkgAesonTH.hs
+++ b/serialise/bench/versus/Macro/PkgAesonTH.hs
@@ -8,12 +8,6 @@ import Data.Aeson.TH as Aeson
 import Data.ByteString.Lazy as BS
 import Data.Maybe
 
-serialise :: [GenericPackageDescription] -> BS.ByteString
-serialise pkgs = Aeson.encode pkgs
-
-deserialise :: BS.ByteString -> [GenericPackageDescription]
-deserialise = fromJust . Aeson.decode'
-
 deriveJSON defaultOptions ''Version
 deriveJSON defaultOptions ''PackageName
 deriveJSON defaultOptions ''PackageId
@@ -21,29 +15,35 @@ deriveJSON defaultOptions ''VersionRange
 deriveJSON defaultOptions ''Dependency
 deriveJSON defaultOptions ''CompilerFlavor
 deriveJSON defaultOptions ''License
-deriveJSON defaultOptions ''SourceRepo
 deriveJSON defaultOptions ''RepoKind
 deriveJSON defaultOptions ''RepoType
+deriveJSON defaultOptions ''SourceRepo
 deriveJSON defaultOptions ''BuildType
-deriveJSON defaultOptions ''Library
-deriveJSON defaultOptions ''Executable
-deriveJSON defaultOptions ''TestSuite
-deriveJSON defaultOptions ''TestSuiteInterface
-deriveJSON defaultOptions ''TestType
-deriveJSON defaultOptions ''Benchmark
-deriveJSON defaultOptions ''BenchmarkInterface
-deriveJSON defaultOptions ''BenchmarkType
-deriveJSON defaultOptions ''BuildInfo
 deriveJSON defaultOptions ''ModuleName
 deriveJSON defaultOptions ''Language
-deriveJSON defaultOptions ''Extension
 deriveJSON defaultOptions ''KnownExtension
+deriveJSON defaultOptions ''Extension
+deriveJSON defaultOptions ''BuildInfo
+deriveJSON defaultOptions ''Library
+deriveJSON defaultOptions ''Executable
+deriveJSON defaultOptions ''TestType
+deriveJSON defaultOptions ''TestSuiteInterface
+deriveJSON defaultOptions ''TestSuite
+deriveJSON defaultOptions ''BenchmarkType
+deriveJSON defaultOptions ''BenchmarkInterface
+deriveJSON defaultOptions ''Benchmark
 deriveJSON defaultOptions ''PackageDescription
 deriveJSON defaultOptions ''OS
 deriveJSON defaultOptions ''Arch
-deriveJSON defaultOptions ''Flag
 deriveJSON defaultOptions ''FlagName
+deriveJSON defaultOptions ''Flag
+deriveJSON defaultOptions ''Condition
 deriveJSON defaultOptions ''CondTree
 deriveJSON defaultOptions ''ConfVar
-deriveJSON defaultOptions ''Condition
 deriveJSON defaultOptions ''GenericPackageDescription
+
+serialise :: [GenericPackageDescription] -> BS.ByteString
+serialise pkgs = Aeson.encode pkgs
+
+deserialise :: BS.ByteString -> [GenericPackageDescription]
+deserialise = fromJust . Aeson.decode'

